packages/edit-site/src/components/navigation-sidebar/navigation-panel/search-results.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { __experimentalNavigationGroup as NavigationGroup } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { normalizedSearch } from './utils';
+import { useSelect } from '@wordpress/data';
+
+export default function SearchResults( { items, search, renderItem } ) {
+	const templateInfos = useSelect(
+		( select ) => {
+			const { __experimentalGetTemplateInfo: getTemplateInfo } = select(
```

Comment:
This is supposed to only work for templates, not for parts (what if a part had slug `index`? 🙂 ).

I guess we could check the first `item`'s type, and if it's a template, proceed to retrieve their info into an otherwise empty array.

E.g. (very simplified and untested!!)
```jsx
const templateType = item[0].type;

const templatesInfo = useSelect( select =>
  'wp_template' === wp_template ? items.map() : []
);

const filtered = useMemo( () => {
  return items.filter( item => {
    if ( 'wp_template' !== templateType ) {
      return normalizedSearch( item.slug, search );
    }

    const { title } = templateInfos.find();
    return normalizedSearch( title, search );
  } );
}, [ items, search, templateType ] );
```

Reply:
Gotcha, totally missed that.